src/commands/createNewProject/ProjectCreateStep/DotnetProjectCreateStep.ts

```diff
@@ -38,7 +38,8 @@ export class DotnetProjectCreateStep extends ProjectCreateStepBase {
         // currentely the version created by func init is behind the template version
         if (context.containerizedProject) {
             const runtime = context.workerRuntime?.capabilities.includes('isolated') ? 'dotnet-isolated' : 'dotnet';
-            await cpUtils.executeCommand(ext.outputChannel, context.projectPath, "func", "init", "--worker-runtime", runtime, "--docker");
+            const targetFramework = runtime === 'dotnet' ? '' : "--target-framework " + nonNullValueAndProp(context.workerRuntime, 'targetFramework');
```

Comment:
Maybe a little comment to say that `target-framework` is only supported with isolated?